### PR TITLE
Large-scale refactoring

### LINKED ISSUE #45: Should work in a pure SVG JSDom context
I'm attempting to use g-chartframe within a JSDom context and getting back weird errors about non-whitespace characters. All this might require is removing save-svg-as-png out of the lib.

### LINKED ISSUE #52: Weird data binding issue
I've noticed that `plot()` returns a selection with data already bound, likely due to [this line](https://github.com/Financial-Times/g-chartframe/blob/4c775e7197247e5dd7ee561f7e22a42e26420bcd/src/chartframe.js#L287). Why data is being bound here I have no idea, but it makes animation really difficult, particularly with deeply nested structures like the gradients I'm using in the Sankey layout.

Truly, most of the data binding done in chartframe.js is unnecessary — you can just append an element and assign an attribute, binding an array of with one element in this context seems like an anti-pattern.

Alas I'm a little bit loth to change it because of how widely used and untested this code is...

### LINKED ISSUE #57: There is no U in "color"
As per ft-interactive/g-chartcolour#20, we should deprecate all APIs that spell "color" with a U.

### PR DESCRIPTION
**Pull request by aendrew**
This does a llllllllot.

1. Upgrades to Babel 7
2. Removes all unnecessary data joins
3. Removes a lllllllot of superfluous code
4. Removes the watermark `g` element if no watermark is present (breaking change)
5. Renames all instances of `colour` to `color` (breaking change)
6. Ensures each element is only created once even if frame is called multiple times (possibly breaking change; element values must be set before frame is first called).

Fixes #57; fixes #52; fixes #45.

### CODE REVIEW

**Comment by aendrew** [comment]
Coverage is now ≅99%.

**Comment by aendrew** [comment]
Updated and merged the "pure SVG context" PR into this one.

File: readme.md [hunk 1]
```diff
@@ -99,12 +99,12 @@ A convenience setter: Set as many values as you choose by passing a big object t
 
 This get the frame to try and automatically comput its top margin. _Not really reccomended though it it used by the pre-packaged webframes_.
 
-<a id="frame-backgroundColour" href="#frame-backgroundColour">#</a>frame.**backgroundColour(_string_)**
+<a id="frame-backgroundColor" href="#frame-backgroundColor">#</a>frame.**backgroundColor(_string_)**
 
 Set the background colour of the frame. For the single argument you can use the same css color naming schemes that you use in HTML, whether that's color names (that is red), rgb values (that is rgb(255,0,0)), hex values, rgba values, etc. If no argument is specified returns the current value.
```
Comment (joannaskao) [inline readme.md]: Let's change `color` in  "background colour of the frame" to `color` for consistency.

**Comment by joannaskao** [comment]
For consistency, I think we should take out this line: https://github.com/Financial-Times/g-chartframe/pull/63/files#diff-61d4073fd0f4d0f39eb017969a6ab5a8R10

**Pull request merged**